tests/pawn-shop.ts

```diff
@@ -189,45 +253,28 @@ describe("PawnHub", () => {
       );
       assert.isTrue(desiredTerms?.duration.eq(TERMS_VALID.duration));
       // These terms are only set when the loan is underwritten.
-      assert.isNull(pawnLoan.terms);
+      assert.isNull(pawnLoanState.terms);
     });
 
-    it("Moves NFT to program escrow", async () => {
-      const pawnTokenAccount = findProgramAddressSync(
-        [
-          Buffer.from("pawn-token-account"),
-          pawnLoanKeypair.publicKey.toBuffer(),
-        ],
-        program.programId
-      )[0];
-
-      const pawnTokenAccountInfo =
-        await program.provider.connection.getAccountInfo(pawnTokenAccount);
-
-      // Before: Mint is in borrower pawn token account.
-      assert.strictEqual(
-        (
-          await pawnMint.getAccountInfo(borrowerPawnTokenAccount)
-        ).amount.toNumber(),
-        1
-      );
-      // Before: Pawn token account not created yet.
-      assert.isNull(pawnTokenAccountInfo);
-
-      await requestLoan(
+    it("Freezes NFT into its original account under the pawn loan's control", async () => {
+      const { pawnTokenAccount, pawnLoan } = await requestLoan(
         program,
-        pawnLoanKeypair,
+        baseKeypair,
         BORROWER_KEYPAIR,
-        pawnMint.publicKey,
         borrowerPawnTokenAccount,
+        pawnMint.publicKey,
         TERMS_VALID
       );
 
-      // After: Mint moved to program escrow.
-      assert.strictEqual(
-        (await pawnMint.getAccountInfo(pawnTokenAccount)).amount.toNumber(),
-        1
+      const pawnTokenAccountInfo =
+        await program.provider.connection.getAccountInfo(pawnTokenAccount);
+
+      const decodedPawnTokenAccountInfo = deserializeTokenAccountInfo(
+        pawnTokenAccountInfo?.data
       );
+
+      assert.isTrue(decodedPawnTokenAccountInfo?.isFrozen);
```

Comment:
somewhat superfluous but should we unit test that once frozen is set, transfers will throw?

Reply:
I don't think so, those are SPL token properties.
Would be equivalent to testing "once the token is i my wallet, can I actually transfer it to someone else or can someone else transfer it on my behalf?"